Opserver.Core/Data/SQL/SQLInstance.Databases.cs

```diff
@@ -99,6 +100,152 @@ public partial class SQLInstance
+            {
+                _type = typeof(T);
+                _properties = GetSettableProps(_type);
+            }
+
+            public SqlMapper.IMemberMap GetMember(string columnName)
+            {
+                columnName = columnName.Replace(" ", "");
+                var property = _properties.Find(p => string.Equals(p.Name, columnName, StringComparison.Ordinal))
+                               ?? _properties.Find(p => string.Equals(p.Name, columnName, StringComparison.OrdinalIgnoreCase));
+
+                if (property == null )
+                {
+                    property = _properties.Find(p => string.Equals(p.Name, columnName.Replace("_", ""), StringComparison.Ordinal))
+                               ?? _properties.Find(p => string.Equals(p.Name, columnName.Replace("_", ""), StringComparison.OrdinalIgnoreCase));
+                }
+
+                if (property != null)
+                    return new SimpleMemberMap(columnName, property);
+
+
+                return null;
+            }
+
+            static List<PropertyInfo> GetSettableProps(Type t)
+            {
+                return t
+                    .GetProperties(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+                    .Where(p => GetPropertySetter(p, t) != null)
+                    .ToList();
+            }
+            
+            static MethodInfo GetPropertySetter(PropertyInfo propertyInfo, Type type)
+            {
+                if (propertyInfo.DeclaringType == type) return propertyInfo.GetSetMethod(true);
+
+                return propertyInfo.DeclaringType.GetProperty(
+                    propertyInfo.Name,
+                    BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance,
+                    Type.DefaultBinder,
+                    propertyInfo.PropertyType,
+                    propertyInfo.GetIndexParameters().Select(p => p.ParameterType).ToArray(),
+                    null).GetSetMethod(true);
+            }
+
+             System.Reflection.ConstructorInfo SqlMapper.ITypeMap.FindConstructor(string[] names, Type[] types) => _type.GetConstructor(new Type[0]);
+
+            ConstructorInfo SqlMapper.ITypeMap.FindExplicitConstructor() => null;
+
+            SqlMapper.IMemberMap SqlMapper.ITypeMap.GetConstructorParameter(
+                System.Reflection.ConstructorInfo constructor, string columnName) => null;
+
+            class SimpleMemberMap : SqlMapper.IMemberMap
+            {
+
+                public SimpleMemberMap(string columnName, PropertyInfo property)
+                {
+                    ColumnName = columnName ?? throw new ArgumentNullException(nameof(columnName));
+                    Property = property ?? throw new ArgumentNullException(nameof(property));
+                }
+
+                public string ColumnName { get; }
+
+                public Type MemberType => Field?.FieldType ?? Property?.PropertyType ?? Parameter?.ParameterType;
```

Comment:
Niggle and it really doesn't matter given how often this is called, but would it make sense to just initialise this in the constructor to avoid recomputing multiple times :P

Reply:
I did have a better version of this thing then tried the CustomAttribute one then went back, but didn't have a commit with it so this one got stuck with some inefficiencies